### ISSUE TITLE
MPI processor support

### ISSUE THREAD

**Issue by streichler** [StanfordLegion/legion#655]
For interoperation with existing MPI libraries, we need a way for Legion applications to run tasks on an "MPI processor".  My initial thinking is that an MPI processor would correspond to a communicator and the task would be run on all the ranks in that communicator.  The processor would have affinity to the system memories on all ranks included in the (sub)communicator, and it would be the problem of the caller/callee to put the right subregions in the right order in the task requirements.  Alternately, this might be a use case for asking for a region requirement on an entire partition and having the mapper select instances for each subregion of the partition.

**Comment by ipdemes** [comment]
@streichler : would you implement some specific Legion communicator class for this?

**Comment by streichler** [comment]
No, the idea would be that, at Realm initialization-time, the application provides one or more MPI communicators for which Realm processors will be created.  A task running on an MPI processor would be able to ask what its communicator is use that.  It can also create and use subcommunicators within a task if it needs to.  All operations on/to/with MPI communicators will be handled by the MPI library, not Realm or Legion.

**Comment by ipdemes** [comment]
OK, that makes sense. I will try to come up with a good use case example for this.
